nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/core/im/IMImageUtils.java

```diff
@@ -124,6 +124,11 @@ protected void makeFiles(Blob blob, String targetExt)
                 targetExt = ext;
             }
             targetFile = Framework.createTempFile("nuxeoImageTarget", "." + targetExt);
+
+            Framework.trackFile(targetFile, targetFile);
```

Comment:
`targetFile` is already tracked in the `call()` method, so why do we need to track it again here?

Reply:
_It's not clear to me exactly why the code needs to change and what's wrong in the original code. Please detail the steps that lead to a leaked undeleted/untracked file._ -> the fix is related to this ticket [NXP-27650](https://jira.nuxeo.com/browse/NXP-27650).  in this ticket they said that _Expected behavior: the temporary must be deleted after the transformation_ and the file are not deleted perhaps it because the GC didn't occurs or reclaims this one.

_targetFile is already tracked in the call() method, so why do we need to track it again here?_ : i added it to the second method in a case where we call the second one `makeFiles` without using the `call(Blob blob, String targetExt, String commandName)` method, i agree is perhaps not clear to did it in this second method and the first one.  and the second method is protected, perhaps i should just keep the tracking  in the second one only near the file creation.

Reply:
> in a case where we call the second one makeFiles

But we don't, all the (non-test) usage of this `IMImageUtils` are from the class itself, the 4 methods crop/resize/rotate/convertToPDF.

We must understand what is behavior is wrong (and reproduce it to be sure) before writing a proposed fix. So here we must understand what sequence of calls leads to a leaked temporary file.

Reply:
yes i see and i agree that this method it's not called in Nuxeo code, but i meat what if a client code call it by adding code directly in Nuxeo and they have they own repo, which i think it's a rare case and i give many supposition :)

The thing is it's not easy to reproduce the problem (the files are not removed after transformations) we related to the GC, then starting from this point i did a reverse engineering and i found this method and i added the tracking but this is confused to have the two method tracking the same file, what i can propos is: 

- Keep the code in the `protected` method, the tracking will occurred near the file creation and i remove those in `call` method.

- I revert this modification, because i agree with you we have already the tacking, but the file wasn't removed, perhaps when the support team test this case the GC didn't reclaim this object or GC didn't started

Reply:
Ok with doing the tracking at `targetFile` creation time (with the file itself as the marker, it shouldn't change GC reachability issues).

But this is just moving code around, there is no change in behavior so the same bug (or not) will be present...